core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -694,7 +694,25 @@ public void onClosed(Channel c, IOException cause) {
         old = ACL.impersonate(ACL.SYSTEM);
         try {
             for (ComputerListener cl : ComputerListener.all()) {
-                cl.onOnline(this,taskListener);
+                try {
+                    cl.onOnline(this,taskListener);
+                } catch (IOException | InterruptedException e) {
```

Comment:
It's typically a good idea to set the interrupt flag on the current thread after catching an interrupted exception.

Reply:
@jvz 
Hm, I didn't know that.  Thanks!  

In other cases I would make the change you suggested, but the point here is to log any exceptions and otherwise ignore them if at all possible. 
 
I don't think we don't want to mark the thread as interrupted here, but I'm open to being wrong.  Can you say a bit more about this?

Reply:
Basically, if you're catching that exception, it's because you were already interrupted, so any IO being cancelled was already cancelled. When you go to set `Thread.currentThread().interrupt()` a second time, it just sets the interrupt status back that was cleared upon catching the exception. Unless this is the end point of the thread at which point it doesn't matter in practice since the thread is about to finish executing.

Typical usage from the other side is something like: `if (Thread.interrupted()) throw new InterruptedException();` during a wait loop of some sort. The if check clears out the interrupted status. By not resetting the interrupt status, that could potentially lead to some sort of deadlock type scenario where, say, an interruptible call to `wait()` is never interrupted because it doesn't know the flag was set.

I originally learned this from an automatic IntelliJ code cleanup suggestion.

TL;DR: multithreading is complicated